main/src/ca/uwaterloo/flix/language/phase/unification/set/SetUnification.scala

```diff
@@ -114,16 +115,16 @@ object SetUnification {
     }
 
     sveAll(state.eqs) match {
-      case None =>
-        // Failure: We found a conflict.
-        // Mark all equations as unsolvable and drop the partial substitution.
-        state.eqs = state.eqs.map(_.toUnsolvable)
-        state.subst = SetSubstitution.empty
-      case Some((_, s)) =>
+      case Result.Ok(s) =>
         // Success: We solved all equations using SVE.
         // Mark all equations as solved and update the substitution.
         state.eqs = Nil
         state.subst = s @@ state.subst
+      case Result.Err(eqs) =>
+        // Failure: We found a conflict.
+        // Store the unsolveable equations and reset the partial substitution.
+        state.eqs = eqs
+        state.subst = SetSubstitution.empty
```

Comment:
What is the purpose of resetting the substitution?

I.e., what is the invariant relating state.eqs, state.subst, and the input?

Reply:
I cannot establish the proper invariant yet.

Reply:
It will need to be a follow up: https://github.com/flix/flix/issues/9870

Reply:
in this branch the system is unsolvable so what does incorrect mean?

Reply:
My current understanding is that -- somehow -- the previous simplification rules before SVE produce subst. that do not respect rigidity. But I need to confirm that.